shared/health/src/main/java/io/pravega/shared/health/StatusAggregationRule.java

```diff
@@ -0,0 +1,75 @@
+/**
+ * Copyright (c) Dell Inc., or its subsidiaries. All Rights Reserved.
+ * <p>
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ * <p>
+ * http://www.apache.org/licenses/LICENSE-2.0
+ */
+package io.pravega.shared.health;
+
+import java.util.Collection;
+
+/**
+ * Any component that implements {@link HealthIndicator} may depend on one or more contributors to determine
+ * its own {@link Status}. An aggregation rule gives components flexibility in determining it's overall health,
+ * given the health of itself and its dependencies.
+ *
+ * Particularly in the case where a component has many instances of a particular service, it may tolerate a certain
+ * number of failures before entering an unhealthy state.
+ *
+ * A {@link Status} will only reduce to an {@link Status#UNKNOWN} state *if* it has no dependencies. A {@link HealthContributor}
+ * that has dependencies signifies that it is now provided with something that (in the abstract) should provide information
+ * about it's {@link Health}.
+ */
+public class StatusAggregationRule {
+    /**
+     * Returns a healthy {@link Status} if a majority of the statuses are healthy.
+     *
+     * @param statuses The statuses to aggregate.
+     * @return A {@link Status} object describing the aggregate status.
+     */
+    public static Status majority(Collection<Status> statuses) {
+        if (statuses.isEmpty()) {
+            return Status.UNKNOWN;
+        }
+        // Should be a strict majority.
+        if (statuses.stream().filter(Status::alive).count() > Math.floor(statuses.size() / 2.0)) {
```

Comment:
It might be simpler to do a for loop and count the alive and other ones, then return their comparison result. Definitely more efficient than creating a `Stream` and doing a division.